sale_order_type/models/sale_order_type.py

```diff
@@ -34,10 +34,10 @@ def default_picking_policy(self):
         comodel_name='account.journal', string='Billing Journal',
         domain=[('type', '=', 'sale')])
     warehouse_id = fields.Many2one(
-        comodel_name='stock.warehouse', string='Warehouse', required=True)
+        comodel_name='stock.warehouse', string='Warehouse')
```

Comment:
You can remove this one, as the company is determined as a related field of this one.